pkg/webui/components/form/index.js

```diff
@@ -148,6 +151,48 @@ class Form extends React.PureComponent {
     validateOnMount: false,
     validateOnChange: false,
     validationSchema: undefined,
+    validationContext: {},
+    validateSync: true,
+  }
+
+  @bind
+  validate(values) {
+    const { validationSchema, validationContext, validateSync } = this.props
+
+    if (!validationSchema) {
+      return {}
+    }
+
+    if (validateSync) {
+      try {
+        validateYupSchema(values, validationSchema, validateSync, validationContext)
+
+        return {}
+      } catch (error) {
+        if (error.name === 'ValidationError') {
+          return yupToFormErrors(error)
+        }
+
+        throw error
+      }
+    }
+
+    return new Promise((resolve, reject) => {
+      validateYupSchema(values, validationSchema, validateSync, validationContext).then(
+        () => {
+          resolve({})
+        },
+        error => {
+          // Resolve yup errors, see https://jaredpalmer.com/formik/docs/migrating-v2#validate.
+          if (error.name === 'ValidationError') {
+            resolve(yupToFormErrors(error))
+          } else {
+            // Throw any other errors as it is not related to the validation process.
+            reject(error)
+          }
+        },
+      )
+    })
```

Comment:
Isn't it possible to do this using async/await? I'd prefer that since it would be more in line with how we usually do async code.